src/django/api/views.py

```diff
@@ -1200,6 +1223,78 @@ def update_location(self, request, pk=None):
         facility_data = FacilityDetailsSerializer(facility).data
         return Response(facility_data)
 
+    @waffle_switch('facility_history')
+    @action(detail=True, methods=['GET'],
+            permission_classes=(IsRegisteredAndConfirmed,),
+            url_path='history')
+    def get_facility_history(self, request, pk=None):
+        """
+        Returns the history of changes to a facility as a list of dictionaries
+        describing the changes.
+
+        ### Sample Response
+            [
+                {
+                    "updated_at": "2019-09-12T02:43:19Z",
+                    "action": "DELETE",
+                    "detail": "Deleted facility"
+                },
+                {
+                    "updated_at": "2019-09-05T13:15:30Z",
+                    "action": "UPDATE",
+                    "changes": {
+                        "location": {
+                            "old": {
+                                "type": "Point",
+                                "coordinates": [125.6, 10.1]
+                            },
+                            "new": {
+                                "type": "Point",
+                                "coordinates": [125.62, 10.14]
+                            }
+                        }
+                    },
+                    "detail": "FacilityLocation was changed"
+                },
+                {
+                    "updated_at": "2019-09-02T21:04:30Z",
+                    "action": "MERGE",
+                    "detail": "Merged with US2019123AG4RD"
+                },
+                {
+                    "updated_at": "2019-09-01T21:04:30Z",
+                    "action": "CREATE",
+                    "detail": "Facility was created"
+                }
+            ]
+        """
+        try:
+            # check to ensure that the endpoint sends back a 404
+            # for facilities which have never existed
+            # Facility.history.get(id=pk)
+
+            facility_history = create_facility_history_list(
+                Facility.history.filter(id=pk),
+                pk,
+            )
+
+            if len(facility_history) == 0:
+                raise NotFound()
+
+            return Response(facility_history)
+        except HistoricalFacility.DoesNotExist:
+            raise NotFound()
+        except:  # noqa
+            try:
+                Facility.history.get(id=pk)
+            except HistoricalFacility.DoesNotExist:
+                raise NotFound()
+
+            _report_facility_history_request_error_to_rollbar(pk)
+            raise APIServerErrorException(
+                'An error prevented responding to the request'
+            )
```

Comment:
I'll test it out tomorrow. The other place where we explicitly log to rollbar here is when attempting to send emails after saving a facility claim, which we do in a try block which catches all exceptions and logs before allowing the rest of the operation to continue. I had thought that the logic here was similar, since this last block is set up to catch all exceptions. However, it may be that we can just remove try altogether and get the 500 "for free", albeit in an unhandled way.

Reply:
c6e6e3476336253e260d443a57dee877f93b5b86
replaces # noqa on a bare except with `except Exception:`